samples/graph/consumer/src/main.rs

```diff
@@ -0,0 +1,263 @@
+    })
+    .await?
+    .into_inner();
+
+    Ok(find_vehicle_response)
+}
+
+/// Get an instance.
+///
+/// # Arguments
+/// * `client` - The digital twin graph client.
+/// * `instance_id` - The instance id.
+/// * `member_path` - The member path.
+async fn get(
+    client: DigitalTwinGraphClient<tonic::transport::Channel>,
+    instance_id: String,
+    member_path: String,
+) -> Result<GetResponse, String> {
+    let retry_strategy = ExponentialBackoff::from_millis(BACKOFF_BASE_DURATION_IN_MILLIS)
+        .map(jitter) // add jitter to delays
+        .take(MAX_RETRIES);
+
+    let request = GetRequest { instance_id: instance_id.clone(), member_path: member_path.clone() };
+
+    let get_response = Retry::spawn(retry_strategy.clone(), || async {
+        let mut client = client.clone();
+
+        client
+            .get(request.clone())
+            .await
+            .map_err(|err_msg| format!("Unable to get the instance due to: {err_msg}"))
+    })
+    .await?
+    .into_inner();
+
+    Ok(get_response)
+}
+
+/// Invoke an instance's operation.
+///
+/// # Arguments
+/// * `client` - The digital twin graph client.
+/// * `instance_id` - The instance id.
+/// * `member_path` - The member path.
+/// * `request_payload` - The request payload.
+async fn invoke(
+    client: DigitalTwinGraphClient<tonic::transport::Channel>,
+    instance_id: String,
+    member_path: String,
+    request_payload: String,
+) -> Result<InvokeResponse, String> {
+    let mut client = client.clone();
+
+    let request = InvokeRequest {
+        instance_id: instance_id.clone(),
+        member_path: member_path.clone(),
+        request_payload: request_payload.clone(),
+    };
+
+    let invoke_response = client
+        .invoke(request.clone())
+        .await
+        .map_err(|err_msg| format!("Unable to invoke the instance's operation due to: {err_msg}"))?
+        .into_inner();
+
+    Ok(invoke_response)
+}
+
+/// Perform a series of interactions with a vehicle digital twin.
+///
+/// # Arguments
+/// * `invehicle_digital_twin_uri` - The in-vehicle digital twin uri.
+async fn interact_with_digital_twin(invehicle_digital_twin_uri: String) -> Result<(), String> {
+    // Connect to the digital twin graph service.
+    let client: DigitalTwinGraphClient<tonic::transport::Channel> =
+        connect_to_digital_twin_graph_service(invehicle_digital_twin_uri.clone()).await?;
+
+    // Find all vehicle instances.
+    let find_vehicle_response: FindResponse =
+        find(client.clone(), sdv::vehicle::ID.to_string()).await?;
+    if find_vehicle_response.values.is_empty() {
+        return Err("Unable to find vehicle instances".to_string());
+    }
+    // For now, we will just use the first vehicle instance.
+    let vehicle: sdv::vehicle::TYPE =
+        serde_json::from_str(&find_vehicle_response.values[0]).unwrap();
+    info!("The vehicle's instance id is: {}", vehicle.instance_id);
+
+    // Get the cabin instance id.
```

Comment:
Maybe extract each action into their own function to reduce the number of lines for this function.

`Get cabin instance`, `get seat instance`, `get seat massager instance`, and `invoke the perform step operation`.

Reply:
Done.